lib/matplotlib/testing/conversion_cache.py

```diff
@@ -0,0 +1,200 @@
+
+        Parameters
+        ----------
+        original : str
+            Full path to the original file.
+        converted : str
+            Full path to the png file converted from the original.
+        """
+        hash_value = self._get_file_hash(original)
+        cached_file = os.path.join(self.cachedir, hash_value + self.cached_ext)
+        with cbook.Locked(self.cachedir):
+            shutil.copyfile(converted, cached_file)
+
+    def _get_file_hash(self, path, block_size=2 ** 20):
+        if path in self.hash_cache:
+            return self.hash_cache[path]
+        md5 = hashlib.md5()
+        with open(path, 'rb') as fd:
+            while True:
+                data = fd.read(block_size)
+                if not data:
+                    break
+                md5.update(data)
+        _, ext = os.path.splitext(path)
+        version_tag = self.converter_version.get(ext)
+        if version_tag:
+            md5.update(version_tag)
+        else:
+            warnings.warn(("Don't know the external converter for %s, cannot "
+                           "ensure cache invalidation on version update.")
+                          % path)
+
+        result = md5.hexdigest()
+        self.hash_cache[path] = result
+        return result
+
+    def report(self):
+        """Return information about the cache.
+
+        Returns
+        -------
+        r : dict
+            `r['gets']` is the set of files queried,
+            `r['hits']` is the set of files found in the cache
+        """
+        return dict(hits=self.hits, gets=self.gets)
+
+    def expire(self):
+        """Delete cached files until the disk usage is under the limit.
+
+        Orders files by access time, so the least recently used files
+        get deleted first.
+        """
+        with cbook.Locked(self.cachedir):
+            stats = {filename: os.stat(os.path.join(self.cachedir, filename))
+                     for filename in os.listdir(self.cachedir)
+                     if filename.endswith(self.cached_ext)}
+            usage = sum(f.st_size for f in stats.values())
+            to_free = usage - self.max_size
+            if to_free <= 0:
+                return
+
+            files = sorted(stats.keys(),
+                           key=lambda f: stats[f].st_atime,
+                           reverse=True)
+            while to_free > 0:
+                filename = files.pop()
+                os.remove(os.path.join(self.cachedir, filename))
+                to_free -= stats[filename].st_size
+
+    @staticmethod
+    def get_cache_dir():
+        cachedir = _get_cachedir()
+        if cachedir is None:
+            raise CacheError('No suitable configuration directory')
+        cachedir = os.path.join(cachedir, 'test_cache')
+        return cachedir
+
+    def ensure_cache_dir(self):
+        if not os.path.exists(self.cachedir):
+            try:
+                cbook.mkdirs(self.cachedir)
+            except IOError as e:
```

Comment:
I think that if os.makedirs raises an OSError, and not an IOError. (cbook.mkdirs depends on os.makedirs, but is untested.)